activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -103,6 +103,16 @@ def index_name_for_remove(table_name, options = {})
       end
 
       class V5_0 < V5_1
+        def create_table(table_name, options = {})
+          # Since version 5.1 Postgres adapter uses bigserial type for primary
+          # keys by default. This compat layer makes old migrations utilize
+          # serial type instead, the way it used to work before 5.1
+          if connection.adapter_name == "PostgreSQL"
```

Comment:
Yeah, I can make that change.

On Thu, Aug 25, 2016 at 10:13 AM, Sean Griffin <notifications@github.com>
wrote:

> In activerecord/lib/active_record/migration/compatibility.rb
> <https://github.com/rails/rails/pull/26266#discussion_r76285358>:
>
> > @@ -103,6 +103,16 @@ def index_name_for_remove(table_name, options = {})
> >        end
> >
> >        class V5_0 < V5_1
> > +        def create_table(table_name, options = {})
> > +          # Since version 5.1 Postgres adapter uses bigserial type for primary
> > +          # keys by default. This compat layer makes old migrations utilize
> > +          # serial type instead, the way it used to work before 5.1
> > +          if connection.adapter_name == "PostgreSQL"
>
> This needs to handle MySQL as well. Is it possible for us to implement
> this in an adapter agnostic way?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/26266/files/de485a4379a265218f1bbea900c08f45e39a93fd#r76285358>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIJEYj45MhqMMLNBPIkwwMhoRSdHCBuks5qjc1DgaJpZM4JrgwZ>
> .
>